src/errors/ResponseError.js

```diff
@@ -1,20 +1,75 @@
 // @flow
 
-class ResponseError extends Error {
-  response: Response
+export type RequestOptionsType = { method: 'GET' } | { method: 'POST', body: FormData }
+
+type ResponseErrorParamsType = {|
+  endpointName: string,
+  response: Response,
+  url: string,
+  requestOptions: RequestOptionsType
+|}
 
-  getMessage = (endpointName: string, response: Response): string =>
-    `ResponseError: Failed to load the request for the ${endpointName} endpoint. Response status ${response.status}`
+class ResponseError extends Error {
+  _endpointName: string
+  _response: Response
+  _url: string
+  _requestOptions: RequestOptionsType
+  _message: string
 
-  constructor (params: { endpointName: string, response: Response }) {
+  constructor (params: ResponseErrorParamsType) {
     super()
 
     if (Error.captureStackTrace) {
       Error.captureStackTrace(this, ResponseError)
     }
 
-    this.response = params.response
-    this.message = this.getMessage(params.endpointName, params.response)
+    const { endpointName, response, url, requestOptions } = params
+
+    this._message = this.createMessage(params)
+    this._endpointName = endpointName
+    this._response = response
+    this._url = url
+    this._requestOptions = requestOptions
+  }
+
+  createMessage ({ requestOptions, url, endpointName, response }: ResponseErrorParamsType): string {
+    const stringifyFormData = (formData: FormData) => {
+      const entries = {}
+      for (const entry of formData.entries()) {
+        entries[entry[0]] = entry[1]
+      }
+
+      return ` and the formData ${JSON.stringify(entries)}`
+    }
+
+    const stringifiedFormData = requestOptions.method === 'POST' ? stringifyFormData(requestOptions.body) : ''
+
+    return `ResponseError: Failed to ${requestOptions.method} the request for the ${endpointName} endpoint with the url
+     ${url}${stringifiedFormData}. Received response status ${response.status}: ${response.statusText}.`
```

Comment:
I think this error message will look weird as a stringified json object will be appended to an `url`

Reply:
Have a look on the code above, I add a string in between: https://github.com/Integreat/integreat-api-client/pull/31/files/f961bab0f5af06cd6c3fdaeb51b2f20c51b1e24b#diff-29f4abd94ac4c57fa229c029b3ea7e6eR42